stdlib/atomic.mli

```diff
@@ -29,6 +29,11 @@ type !'a t
 (** Create an atomic reference. *)
 val make : 'a -> 'a t
 
+(** Create a cache line-aligned atomic reference. Use it over [make v] to
+    prevent false-sharing. Atomic reference allocated with [make_aligned v]
+    occupies 8-16x the memory of one allocated with [make v]. *)
```

Comment:
Reading the PR again, this is indeed `8-16x`, as 64B alignment is not used currently in the PR. Sorry for the noise.

Reply:
> cache line-aligned -> cache-line-aligned

Opinions from native English speakers would be welcome.  A Google search shows occurrences of "cache line aligned", "cache-line aligned", and "cache line-aligned", but not "cache-line-aligned".

(I guess there's a meta-rule that you can use at most one hyphen in a compound adjective.)  

As a non-native speaker, I'd rather chicken out and write "Create an atomic reference that is aligned on a cache line.".

Reply:
"Create an atomic reference that is aligned on a cache line" is indeed the clearest of those options.

Reply:
Thanks, much better indeed. Just to clarify, after referring to the Chicago Manual of Style, I stand by "cache line-aligned" being incorrect and "cache-line-aligned" being correct as a form derived from "cache line aligned". Whether the latter phrase sounds entirely right to a native speaking is something else, but it is not a matter of a limit on the number of hyphens one can use. You will find more instances of "cache-line-aligned" on Google scholar.

Reply:
> Opinions from native English speakers would be welcome. A Google search shows

This is so 2022...

GPT-4 says:

> The correct usage depends on the style guide that you are following. However, in many English writing contexts, the hyphen is used to link words together to avoid ambiguity. In this particular case, "cache-line-aligned" would be the most appropriate usage because it clearly conveys that the phrase is referring to alignment with respect to cache lines, not an aligned cache or line. The hyphenation creates a single, coherent idea.

> In contrast, "cache line-aligned" could potentially be misread as "cache, line-aligned", which conveys a different meaning. 

> If no specific style guide is being followed, the important thing is to be consistent throughout your text. If you choose to use "cache-line-aligned," make sure you use the same structure throughout your work. 


GPT-3.5 begs to differ:

> The common and widely accepted notation for the term you mentioned is "cache line-aligned," with a space between "cache" and "line." This notation helps clarify that "line-aligned" refers to alignment with the cache line size. It's important to note that style conventions can vary, and different publications or organizations may have their own preferences. However, using "cache line-aligned" is a generally accepted and clear way to express the concept.

Reply:
Thank you all. I've put Anil's suggestion in place.